src/components/Editable/useOnCut.ts

```diff
@@ -0,0 +1,17 @@
+import React from 'react'
+import * as selection from '../../device/selection'
+
+/** Returns an onCopy handler that cuts the selection text, sets a text/em flag in the clipboard data to detect the source on paste and removes the current selection. */
+const useOnCut = (event: React.ClipboardEvent) => {
```

Comment:
As a hook, this should return a callback that takes the ClipboardEvent. It's not really a hook if it is used directly as the event handler itself.